ray-operator/controllers/ray/utils/util.go

```diff
@@ -71,6 +71,40 @@ func IsCreated(pod *corev1.Pod) bool {
 	return pod.Status.Phase != ""
 }
 
+func FindHeadPodReadyCondition(pods corev1.PodList) metav1.Condition {
+	var headPod *corev1.Pod
+	for _, pod := range pods.Items {
+		if pod.Labels[RayNodeTypeLabelKey] == string(rayv1.HeadNode) {
+			headPod = &pod
+			break
+		}
+	}
+
+	replicaHeadReadyCondition := metav1.Condition{
+		Type:   string(rayv1.HeadReady),
```

Comment:
Maybe we can rename it from `HeadReady` to `HeadPodReady` because the upstream `PodConidtion` uses [PodReady](https://pkg.go.dev/k8s.io/api/core/v1#PodConditionType) instead of `Ready` for the condition type.

Reply:
update to `HeadPodReady` with `Ready` as value
https://github.com/ray-project/kuberay/blob/41bd563f3af567efed34bd3181957f1005e842eb/ray-operator/apis/ray/v1/raycluster_types.go#L176